File: Build 'N Crash/Assets/SCRIPTS/HUD/Speed_Script.js
#pragma strict

public var player:GameObject;
public var speed:UI.Text;
var valueOfSpeed:int = 0;
var lastPosition:Vector3;

function Start () {

}

function FixedUpdate () {
speed.text = (valueOfSpeed).ToString();
checkSpeed();
}


function checkSpeed()
{
	valueOfSpeed = (((player.transform.position - lastPosition).magnitude) / Time.deltaTime) ;
     lastPosition = player.transform.position;
}

File: Build 'N Crash/Assets/SCRIPTS/HUD/Speed_Script.cs
using UnityEngine;
using UnityEngine.VR;
using UnityEngine.UI;
using System.Collections;

public class Speed_Script : MonoBehaviour 
{
	public Text speed;
	public Text refreshRate;

	public void Start () {}

	public void FixedUpdate ()
	{
		speed.text = gameObject.GetComponent<ControlScript>().getSpeed().ToString();
		refreshRate.text = VRDevice.refreshRate.ToString();
	}
}
